diff-containers/src/Data/Map/Diff/Strict.hs

```diff
@@ -87,42 +78,14 @@ newtype Diff k v = Diff (Map k (NEDiffHistory v))
 -- A history has an implicit sense of ordering according to time: from left to
 -- right. This means that the leftmost element in the history is the /earliest/
 -- change, while the rightmost element in the history is the /latest/ change.
-newtype UnsafeDiffHistory t v = UnsafeDiffHistory (t (DiffEntry v))
-  deriving stock (Generic, Functor, Foldable)
-
-deriving stock instance Show v => Show (UnsafeDiffHistory Seq v)
-deriving stock instance Show v => Show (UnsafeDiffHistory NESeq v)
-deriving stock instance Eq v => Eq (UnsafeDiffHistory Seq v)
-deriving stock instance Eq v => Eq (UnsafeDiffHistory NESeq v)
-
-{-# COMPLETE DiffHistory #-}
-
-newtype DiffHistory v = MkDiffHistory (UnsafeDiffHistory Seq v)
-  deriving stock (Generic, Show, Eq, Functor)
-  deriving newtype (NoThunks, Foldable)
-
-{-# COMPLETE NEDiffHistory #-}
-
--- | A non-empty diff history.
-newtype NEDiffHistory v = MkNEDiffHistory (UnsafeDiffHistory NESeq v)
-  deriving stock (Generic, Show, Eq, Functor)
-  deriving newtype (NoThunks, Foldable)
-
-pattern DiffHistory :: Seq (DiffEntry v) -> DiffHistory v
-pattern DiffHistory { unDiffHistory } =
-  MkDiffHistory (UnsafeDiffHistory unDiffHistory)
-
-pattern NEDiffHistory :: NESeq (DiffEntry v) -> NEDiffHistory v
-pattern NEDiffHistory { unNEDiffHistory } =
-  MkNEDiffHistory (UnsafeDiffHistory unNEDiffHistory)
+newtype DiffHistory v = DiffHistory { getDiffHistory :: Seq (DiffEntry v) }
```

Comment:
Is it ok to export this constructor?

Reply:
Good point -- I could move these definitions to a `*.Diff.Internal` module, and have `*.Diff` export only the "safe" API. Sounds okay?

Reply:
Sure!